### PR TITLE
Nvrtc

### PR DESCRIPTION
**Pull request by lilkeet**
## What
This pr adds two new modules to the library: _nvrtc_ and _cuda_. 

_cuda_ is better known the cuda driver api, which is very similar to the cuda runtime api but a bit lower level. 

_nvrtc_ is a module that enables the use of runtime compilation of cuda code.

An example of runtime compilation is also added. https://github.com/SciNim/nimcuda/issues/19

## Why

Currently, there is no good way to run cuda kernels from Nim. Here are some known options:

#### Use a cuda compiler to compile the entire project
- nvcc is incompatible with the nim compiler https://github.com/nim-lang/Nim/issues/16936
- cuda enabled clang can work with some setup
- some of nim's c macros interfere with the cuda library's macros (`__no_inline__`), requiring manual editing of the installed cuda headers.
- forces all code in the project to run using cuda

#### Transpile nim code to ptx
- can use an intermediary like llvm https://github.com/mratsim/constantine/blob/077cfb2/constantine/platforms/code_generator/nvidia.nim#L158-L270
- outside dependencies
- lots of setup and intimate knowledge required

This pr enables a native, easy solution. Checkout the example (355290f903e316ca1a7303337a61b5ba9eede3f5) for all that is required to run a cuda kernel. Much of it is boilerplate that could be minimized by a higher-level library.

The cuda driver api was added mostly due to the fact that nvrtc requires that you use it instead of the cuda runtime api, but it should've been added long ago, imo.

### CODE REVIEW

File: nimcuda.nimble [hunk 1]
```diff
@@ -29,26 +29,29 @@ const
...
+  # "curand",
+  # "cusolver_common",
+  # "cusolverDn",
+  # "cusolverRf",
+  # "cusolverSp",
+  # "cusparse",
+  # "nvblas"
+  # #"nvgraph" <- removed in cuda 11.0, adopted into cugraph
+  # "nvrtc",
+  "cuda"
```
Comment (Vindaar) [inline nimcuda.nimble:53]: Why are all others commented, aside from `cuda` out now?

**Comment by Vindaar** [comment]
Aside from my question, thanks a lot! :rocket:

File: nimcuda.nimble (hunk 1 above)
Comment (lilkeet) [reply nimcuda.nimble:53]: I commented those out since i was only generating nvrtc's nim file, then cuda's. forgot to change it back!

Comment (Vindaar) [reply nimcuda.nimble:53]: Will you change it back before I merge?

Comment (lilkeet) [reply nimcuda.nimble]: Changed it back. Thanks for the catch! :) 0ec8f5d92a3d3010ca7d61d707a488baaef1d642

**Comment by Vindaar** [comment]
Thanks a lot again!

**Pull request merged**